File: smartlist/src/main/java/com/smartlist/module/assessment/dto/request/DuplicateAssessmentForOrganizationsReqDTO.java
package com.smartlist.module.assessment.dto.request;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class DuplicateAssessmentForOrganizationsReqDTO {

    @Schema(description = "Assessment Id", example = "1", requiredMode= Schema.RequiredMode.REQUIRED)
    @Positive
    @NotNull
    Integer assessmentId;

    @Schema(description = "List of Organization Ids", example = "1", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    List<Integer> organizationIds;


}

File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/assessment/dto/request/duplicate-assessment-for-organizations.req.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty} from 'class-validator';

export class DuplicateAssessmentForOrganizationsReqDto {
  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  assessment_id: string;

  @ApiProperty({
    type: String,
  })
  organization_ids: string[];
}
